examples/SharedMemory/PhysicsDirect.cpp

```diff
@@ -608,6 +617,57 @@ bool PhysicsDirect::processCamera(const struct SharedMemoryCommand& orgCommand)
 	return m_data->m_hasStatus;
 
 
+}
+
+// NOTE: using pointers to shared memory here, to avoid yet another byte-by-byte memory copy
+bool PhysicsDirect::processCameraArray(const struct SharedMemoryCommand& orgCommand)
+{
+	// printf("PhysicsDirect::processCameraArray\n");
+	SharedMemoryCommand command = orgCommand;
+
+	const SharedMemoryStatus& serverCmd = m_data->m_serverStatus;
+	bool hasStatus = m_data->m_commandProcessor->processCommand(command,m_data->m_serverStatus,&m_data->m_bulletStreamDataServerToClient[0],SHARED_MEMORY_MAX_STREAM_CHUNK_SIZE);
+
+	b3Clock clock;
+	double startTime = clock.getTimeInSeconds();
+	double timeOutInSeconds = m_data->m_timeOutInSeconds;
+
+	while ((!hasStatus) && (clock.getTimeInSeconds()-startTime < timeOutInSeconds))
+	{
+		const  SharedMemoryStatus* stat = processServerStatus();
+		if (stat)
+		{
+			hasStatus = true;
+		}
+	}
+
+	m_data->m_hasStatus = hasStatus;
+	if (hasStatus)
+	{
+		btAssert(m_data->m_serverStatus.m_type == CMD_CAMERA_ARRAY_IMAGE_COMPLETED);
+
+		if (m_data->m_verboseOutput)
+		{
+			b3Printf("Camera array image OK\n");
+		}
+
+		int numBytesPerPixel = 3; //RGB
+		m_data->m_ptrCameraArrayPixelsRGB = (unsigned char*)&(m_data->m_bulletStreamDataServerToClient[0]);
```

Comment:
This is a dodgy place - as I'm not sure, if a pointer suffices. See also a comment above.